src/test/selftest-parallel

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+if test "$NUM_PARTITIONS" = ''; then
+    NUM_PARTITIONS=1
+fi
+if test "$RUN_PARTITIONS" = ''; then
+    RUN_PARTITIONS=$(seq 0 $((NUM_PARTITIONS-1)));
+fi
+if test "$TEST_SPEC" = ''; then
+    TEST_SPEC="~[.]" # All non-hidden tests by default
+fi
+
+cleanup() {
+    rm -rf $TEST_PARTITIONS_DIR
+}
+TEST_PARTITIONS_DIR=$(mktemp -d ./test-partitions.XXXXXXXX)
+trap cleanup 0 2 15
+
+cd $TEST_PARTITIONS_DIR
+perl -e 'my @all_tests = `../stellar-core --ll FATAL --test --list-test-names-only '"$TEST_SPEC"'`;
+         for (my $p = 0; $p < '"$NUM_PARTITIONS"'; $p++) {
+             open(my $out, ">", "test-partition-$p.txt");
+             for (my $i = $p; $i < @all_tests; $i+='"$NUM_PARTITIONS"') {
+                 print $out $all_tests[$i];
+             }
+             close $out;
+         }'
+cd ..
+
+declare -a TEST_JOBS
+for i in $RUN_PARTITIONS; do
```

Comment:
I suspect we need to update the trap handler to also kill processes in `TEST_JOBS` in case the script is interrupted - maybe just kill the group with something like:
```
kill -INT -$$
sleep 2
kill -KILL -$$
```